com.regnosys.rosetta.ide/src/com/regnosys/rosetta/ide/formatting/RosettaFormattingService.java

```diff
@@ -0,0 +1,69 @@
+package com.regnosys.rosetta.ide.formatting;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.lsp4j.FormattingOptions;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.xtext.formatting.IIndentationInformation;
+import org.eclipse.xtext.formatting2.IFormatter2;
+import org.eclipse.xtext.formatting2.regionaccess.ITextReplacement;
+import org.eclipse.xtext.ide.server.Document;
+import org.eclipse.xtext.ide.server.formatting.FormattingService;
+import org.eclipse.xtext.preferences.MapBasedPreferenceValues;
+import org.eclipse.xtext.resource.XtextResource;
+import org.eclipse.xtext.util.TextRegion;
+
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.regnosys.rosetta.formatting2.RosettaFormatterPreferenceKeys;
+
+/**
+ * This class allows passing a `maxLineWidth` parameter
+ * to client requests.
+ */
+public class RosettaFormattingService extends FormattingService {
+	@Inject(optional = true)
+	private Provider<IFormatter2> formatter2Provider;
+
+	@Inject
+	private IIndentationInformation indentationInformation;
+		
+	@Override
+	public List<TextEdit> format(XtextResource resource, Document document, int offset, int length,
+			FormattingOptions options) {
+		String indent = indentationInformation.getIndentString();
+		if (options != null) {
+			if (options.isInsertSpaces()) {
+				indent = Strings.padEnd("", options.getTabSize(), ' ');
+			}
+		}
+		List<TextEdit> result = new ArrayList<>();
+		if (this.formatter2Provider != null) {
+			MapBasedPreferenceValues preferences = new MapBasedPreferenceValues();
+			preferences.put("indentation", indent);
+			
+			Number conditionalMaxLineWidth = options.getNumber("conditionalMaxLineWidth");
```

Comment:
We should use constants for these. I saw some tests referring to these string.

Reply:
Agreed. Most of this code is copied over from the parent class (including the magic string `"indentation"`), I've only made a couple of adjustements. But yeah, it's not because Xtext doesn't do it the right way, that we should, so I'll adjust that.